app/models/TraderProfile.scala

```diff
@@ -31,26 +31,24 @@ object TraderProfile {
 
   implicit lazy val format: OFormat[TraderProfile] = Json.format
 
-  def build(answers: UserAnswers): IorNec[ValidationError, TraderProfile] =
+  def build(answers: UserAnswers): EitherNec[ValidationError, TraderProfile] =
     (
-      answers.getIor(UkimsNumberPage),
+      answers.getEither(UkimsNumberPage),
       getNirms(answers),
       getNiphl(answers)
     ).parMapN(TraderProfile.apply)
 
-  private def getNirms(answers: UserAnswers): IorNec[ValidationError, Option[String]] =
-    answers.getIor(HasNirmsPage).flatMap {
-      case true  =>
-        answers.getIor(NirmsNumberPage).map(Some(_))
-      case false =>
-        if (answers.isDefined(NirmsNumberPage)) Ior.Left(NonEmptyChain.one(UnexpectedPage(NirmsNumberPage))) else Ior.Right(None)
+  private def getNirms(answers: UserAnswers): EitherNec[ValidationError, Option[String]] =
+    answers.getEither(HasNirmsPage) match {
+      case Right(true) => answers.getEither(NirmsNumberPage).map(Some(_))
+      case Right(false) => if(answers.isDefined(NirmsNumberPage)) Left(NonEmptyChain.one(UnexpectedPage(NirmsNumberPage))) else Right(None)
+      case Left(errors) => Left(errors)
     }
 
-  private def getNiphl(answers: UserAnswers): IorNec[ValidationError, Option[String]] =
-    answers.getIor(HasNiphlPage).flatMap {
-      case true  =>
-        answers.getIor(NiphlNumberPage).map(Some(_))
-      case false =>
-        if (answers.isDefined(NiphlNumberPage)) Ior.Left(NonEmptyChain.one(UnexpectedPage(NiphlNumberPage))) else Ior.Right(None)
+  private def getNiphl(answers: UserAnswers): EitherNec[ValidationError, Option[String]] =
+    answers.getEither(HasNiphlPage) match {
+      case Right(true)  => answers.getEither(NiphlNumberPage).map(Some(_))
+      case Right(false)  => if (answers.isDefined(NiphlNumberPage)) Left(NonEmptyChain.one(UnexpectedPage(NiphlNumberPage))) else Right(None)
```

Comment:
we can probably extract this bit to user answers and modify the function name to be more intuitive, maybe something like:

```
def unexpectedValueDefined(page: Gettable[_]):  = (answers.isDefined(NiphlNumberPage)) Left(NonEmptyChain.one(UnexpectedPage(NiphlNumberPage))) else Right(None)


def getPageValue[A](page: Gettable[A])(implicit rds: Reads[A]): EitherNec[ValidationError, A] =
   get(page).map(Right(_)).getOrElse(Left(NonEmptyChain.one(PageMissing(page))))
```